File: Lesson09_OOP/05_Encapsulation/Pants.java
/*
Copyright Ann Barcomb and Emily Marasco, 2022
Licensed under GPL v3
See LICENSE.txt for more information.
*/

public class Pants {
    private Clothing fabric;
    private int waistSize;
    private int legSize;

    public Pants(String colour, String material, int waist, int leg) {
        this.fabric = new Clothing(colour, material);
        this.waistSize = waist;
        this.legSize = leg;
    }

    public Pants(int colour, String material, int waist, int leg) {
        this.fabric = new Clothing(colour, material);
        this.waistSize = waist;
        this.legSize = leg;
    }

    public String getMaterial() { return this.fabric.getMaterial(); }
    public String getColour() { return this.fabric.getColour(); }
    public int getColourID() { return this.fabric.getColourID(); }
    public int getWaistSize() { return this.waistSize; }
    public int getLegSize() { return this.legSize; }

    public void setMaterial(String material) { this.fabric.setMaterial(material); }
    public void setColour(String colour) { this.fabric.setColour(colour); }
    public void setColour(int colour) { this.fabric.setColour(colour); }
    public void setHipSize(int leg) { this.legSize = leg; }
    public void setWaistSize(int waist) { this.waistSize = waist; }

    public static void main(String[] args) {
        var myPants = new Pants(1, "cotton", 34, 36);
        System.out.printf("Colour: %s, Material: %s, Waist: %d, Leg: %d\n", 
          myPants.getColour(), myPants.getMaterial(), myPants.getWaistSize(),
          myPants.getLegSize());
    }
}

File: Lesson09_OOP/05_Encapsulation/Pants.py
"""
Copyright Khawla Shnaikat and ChatGPT 3.5, 2023
Licensed under GPL v3
See LICENSE.txt for more information.
"""
from clothing import Clothing
class Pants:
    def __init__(self, colour, material, waist, leg):
        self.fabric = Clothing(colour, material)
        self.waist_size = waist
        self.leg_size = leg

    def get_material(self):
        return self.fabric.get_material()

    def get_colour(self):
        return self.fabric.get_colour()

    def get_colour_id(self):
        return self.fabric.get_colour_id()

    def get_waist_size(self):
        return self.waist_size

    def get_leg_size(self):
        return self.leg_size

    def set_material(self, material):
        self.fabric.set_material(material)

    def set_colour(self, colour):
        self.fabric.set_colour(colour)

    def set_waist_size(self, waist):
        self.waist_size = waist

    def set_leg_size(self, leg):
        self.leg_size = leg
if __name__ == "__main__":
    my_pants = Pants("black", "denim", 34, 32)
    print(f"Pants Colour: {my_pants.get_colour()}, Material: {my_pants.get_material()}, Waist: {my_pants.get_waist_size()}, Leg: {my_pants.get_leg_size()}")
